packages/rum-core/src/rawRumEvent.types.ts

```diff
@@ -209,3 +209,31 @@ export interface CommonContext {
   context: Context
   hasReplay?: true
 }
+
+export interface ReadonlyLocation {
+  readonly hash: string
+  readonly host: string
+  readonly hostname: string
+  readonly href: string
+  readonly origin: string
+  readonly pathname: string
+  readonly port: string
+  readonly protocol: string
+  readonly search: string
+}
+
+export type RumEventDomainContext<E extends RawRumEvent> = E extends RawRumErrorEvent
+  ? { error?: unknown }
+  : E extends RawRumActionEvent
+  ? { event?: Event }
+  : E extends RawRumLongTaskEvent
+  ? { performanceEntry: PerformanceEntry }
+  : E extends RawRumResourceEvent
+  ? {
+      performanceEntry?: PerformanceEntry
+      xhr?: XMLHttpRequest
+      response?: Response
```

Comment:
I didn't expose the `Request` object in case of `fetch`, because it can't be accessed if it not passed as first argument of the `fetch` function, so this would be rather limited. Also, the most interesting infos of the `Request` object is already in the RUM event. But I could add it if we fill like it would be a good addition.

Reply:
Why not create a request object out of the user fetch proxy params?

Reply:
I think that could works, yeah. It could simplify the [tracer](https://github.com/DataDog/browser-sdk/blob/ca9e9d3998d990fbc0e5829b3318335fc855fd80/packages/rum-core/src/domain/tracing/tracer.ts#L31-L53) too, if we always have a `Request` object instead of various kind of `input/init`.

Maybe we should try this in a separate PR, since this one is wide enough.

Reply:
a couple of suggestions:
- reflect in the naming that this is related to fetch
- we should also handle the `response: Error` case
- what about exposing `{ input: RequestInfo, init?: RequestInit, response: Response | Error  }`?

Reply:
Ok to expose input/init. Ok to expose `error` but I'd like to use a different field if that's ok :)

Reply:
Implemented as `{ fetchResponse?: Response, fetchInput: RequestInfo, fetchInit?: RequestInit, fetchError?: Error }`

Reply:
What is your reasoning for splitting response and error? 
IMO keeping it close to browser behavior could be easier to understand where to get the data

Reply:
It is not the browser behavior. Our implementation receives a value that is either an error or a response, but that's because we use the same function to handle both successful and failed calls. I think having separate properties makes it easier to use, because we can simply ducktype instead of using `instanceof`.

Reply:
oh indeed, great then!